crates/router/src/types/api.rs

```diff
@@ -162,6 +165,42 @@ pub enum ConnectorCallType {
     Single(ConnectorData),
 }
 
+#[derive(EnumString, Serialize, Deserialize, PartialEq, Debug)]
+#[strum(serialize_all = "lowercase")]
+enum ConnectorName {
```

Comment:
@vspecky @Narayanbhat166 Correct me if I'm wrong, I believe we'd prefer to use the [`RoutableConnectors`](https://github.com/juspay/hyperswitch/blob/3d05e50abcb92fe7e6c4472faafc03fb70920048/crates/api_models/src/enums.rs#L655) enum instead of adding a new one with the same enum variants.

Reply:
We would prefer to use the `[Connector](https://github.com/juspay/hyperswitch/blob/3d05e50abcb92fe7e6c4472faafc03fb70920048/crates/api_models/src/enums.rs#LL582C21-L582C21)` enum. Just to re-iterate, `Connector` is more general whereas `RoutableConnectors` is intended for routing purposes and is a direct subset of `Connector`.

Reply:
@nachiketkanore Could you please make the change to use the existing `Connector` enum instead?

Reply:
The ApplePay variant is not implemented inside [connector](https://github.com/juspay/hyperswitch/blob/e5cc0d9d45d41c391720ceb3f6c18151ac5a00f2/crates/router/src/connector.rs)
How do you want to handle [ApplePay](https://github.com/juspay/hyperswitch/blob/3d05e50abcb92fe7e6c4472faafc03fb70920048/crates/api_models/src/enums.rs#L586) and [Dummy](https://github.com/juspay/hyperswitch/blob/3d05e50abcb92fe7e6c4472faafc03fb70920048/crates/api_models/src/enums.rs#L594) for this?

Reply:
@nachiketkanore You can just remove the Applepay variant. It is residual (@Sangamesh26). Other than that, I'm not completely sure if we need both `Dummy` and `DummyConnector` variants. Is there any reason we should have both? @Narayanbhat166 @SanchithHegde

Reply:
We don't need both. `Dummy` was added initially as a default value. Now that the dummy connector is being implemented under the `DummyConnector` variant, it should be fine to remove the `Dummy` variant, and make the `DummyConnector` variant as the default.

Reply:
DummyConnector cannot be made default since it is behind the feature #[cfg(feature = "dummy_connector")]

Reply:
Can you remove the default trait? I don't see it being used anywhere